### PR TITLE
Added the ability to have different aliases for the same commands based on the cli's version

### PR DESCRIPTION
**Pull request by cgotcu**
The arachni version I am using (1.3.2) seems to have different parameters for running xss attacks for example. This allows users to use gauntlt for various attacks without having to modify the code other than the attack_aliases/ json file corresponding to the attack being ran. 

Ideally we would merge the attack_alias json file that comes with gauntlt with one provided by the user for their one-offs and different versions so to keep the user out of the actual gauntlt code.

### CODE REVIEW

**Comment by cgotcu** [comment]
Working on fixing the build (there are some tests unrelated to my commit that seem to be failing as well) and then will try again.

**Comment by apolloclark** [comment]
Cool, I'll take a look at your pull request tomorrow and get you some feedback.

**Comment by apolloclark** [comment]
Cool, I'll check this over and get you some feedback tomorrow. Thanks!

File: .gitignore [hunk 1]
```diff
@@ -28,6 +28,7 @@ Gemfile.lock
 .rbenv*
 *.gem
 .rspec
+.idea
```
Comment (apolloclark) [inline .gitignore]: Change this to ".idea/" to remove the IntelliJ files.

File: lib/gauntlt/attack_adapters/support/attack_step_helper.rb [hunk 2]
```diff
@@ -7,10 +7,25 @@ module AttackAliasHelper
...
+
+	print "Checking #{@alias_versioned} and #{alias_name}"
+
+        if @attack_steps.has_key?(@alias_versioned)
+            @attack_steps[@alias_versioned]
+        elsif @attack_steps.has_key?(alias_name)
+            @attack_steps[alias_name]
+        else
+            raise("Unknown attack alias #{alias_name}, try gauntlt --list_attack_aliases")
+        end
```
Comment (apolloclark) [inline lib/gauntlt/attack_adapters/support/attack_step_helper.rb]: @wickett this looks good, what do you think?

**Comment by apolloclark** [comment]
Yes, this looks like a good pull request. Update the .gitignore to remove the IntelliJ files. I'll test this tomorrow.

**Comment by apolloclark** [comment]
It's nice to have OWASP Zap included, but please add it to .gitignore, we don't want to include a specific version of an optional dependency.

**Comment by cgotcu** [comment]
Ha! I didn't realize after I cancelled the pull request that my feeble attempts at doing ruby things would still show up.

**Comment by apolloclark** [comment]
That's fine, I reopened the pull request, you were on the right track. :) Best way to add OWASP-Zap support is to create a new branch on your forked repo, and open a separate pull request.

**Comment by cgotcu** [comment]
Can we cancel this one again so I can squash the commits and look like less of a noob? :)

**Comment by apolloclark** [comment]
Sure, no problem. Create a new pull request for the Arachni changes, based on this commit: https://github.com/cgotcu/gauntlt/commit/3de6abcb2d1e01158c8480cc9ec6c4615273581a

**Pull request closed without merging**